src/common/factory.cc

```diff
@@ -19,6 +19,9 @@ IndexFactory::Create(const std::string& name, const int32_t& version, const Obje
     static_assert(KnowhereDataTypeCheck<DataType>::value == true);
     auto& func_mapping_ = MapInstance();
     auto key = GetIndexKey<DataType>(name);
+    if (func_mapping_.find(key) == func_mapping_.end()) {
+        LOG_KNOWHERE_ERROR_ << "failed to find index " << key << " in factory";
+    }
```

Comment:
This function returns an `Index` object, and if no mapping is found, then this function has to throw. Alternatively, this function may get changed to return `std::shared_ptr<Index>` or `std::optional<Index>` and return `{}`